pkg/serverless/metrics/metric_test.go

```diff
@@ -106,13 +107,17 @@ func TestRaceFlushVersusAddSample(t *testing.T) {
 
 	assert.NotNil(t, metricAgent.Demux)
 
-	go func() {
-		http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+	server := http.Server{
+		Addr: "localhost:8888",
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			time.Sleep(10 * time.Millisecond)
-		})
+		}),
+	}
+	defer server.Close()
```

Comment:
Changes to this file mean that we can now run this test multiple times without it failing. Prior, the goroutine where the server is running was leaked, it would continue to run even after the test ended. Therefore, a second run of the test would always fail because the port was in use.

Use `go test -count=2 ./pkg/serverless/...` to see this change.